streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -72,8 +82,23 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
         } else {
             throw new InvalidStateStoreException("Cannot get state store " + storeName + " because the stream thread is " +
                                                      state + ", not RUNNING" +
-                                                     (includeStaleStores ? " or REBALANCING" : ""));
+                                                     (storeQueryParams.includeStaleStores() ? " or REBALANCING" : ""));
         }
     }
 
+    private TaskId createKeyTaskId(final String storeName, final Integer partition) {
+        if (partition == null) {
+            return null;
+        }
+        final List<String> sourceTopics = internalTopologyBuilder.stateStoreNameToSourceTopics().get(storeName);
+        final Set<String> sourceTopicsSet = sourceTopics.stream().collect(Collectors.toSet());
+        final Map<Integer, InternalTopologyBuilder.TopicsInfo> topicGroups = internalTopologyBuilder.topicGroups();
+        for (final Map.Entry<Integer, InternalTopologyBuilder.TopicsInfo> topicGroup : topicGroups.entrySet()) {
```

Comment:
this looks right to me.. (I can't be sure unless I step through the code though :))

Reply:
Doing this check in stores() leads to an increase in cyclomatic complexity above 16. So, I have added it here.

Reply:
I have added an integration test covering all 4 cases.

- Fetch a key, by default gets stores for all partitions without stale store. Only active returns the value and standby returns null.

- Fetch a key from a stale store from a specific partition. If the partition is available and active, it returns the key if not it throws InvalidStateStoreException.

- Fetch a key with stale stores enabled, active and standby both should return the value.

- Fetch a key with stale stores and enableda and from a specific partition, active and standby for that partition, both return the value. If you ask the key from a wrong partition, it returns null.